.github/workflows/hosted.yml

```diff
@@ -0,0 +1,290 @@
+      with:
+        version: 13
+        platform: x64
+        cygwin: 0
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Use ccache
+      if: startswith(matrix.os, 'macos') || startswith(matrix.os, 'ubuntu')
+      uses: hendrikmuhs/ccache-action@v1
+      with:
+        key: ${{ matrix.os }}-${{ matrix.compiler }}
+
+    - name: Configure shell (Ubuntu)
+      if: startswith(matrix.os, 'ubuntu')
+      run: echo 'PATH=/usr/lib/ccache:'"$PATH" >> $GITHUB_ENV
+
+    - name: Configure shell (MacOS)
+      if: startswith(matrix.os, 'macos')
+      run: echo "PATH=$(brew --prefix)/opt/ccache/libexec:$PATH" >> $GITHUB_ENV
+
+    - name: Build (Windows)
+      if: startswith(matrix.os, 'windows')
+      shell: cmd
+      run: |
+        call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
+
+        if "${{ matrix.compiler }}" EQU "cl" (
+          echo 'CC=cl' >> $GITHUB_ENV
+          echo 'CXX=cl' >> $GITHUB_ENV
+        ) else (
+          echo 'CC=clang' >> $GITHUB_ENV
+          echo 'CXX=clang++' >> $GITHUB_ENV
+        )
+
+        set
+        where cmake && cmake --version
+        where ninja && ninja --version
+        where %CC% && %CC% -version
+        where %CXX% && %CXX% -version
+
+        cd runtime/Cpp
+
+        cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DANTLR_BUILD_CPP_TESTS=OFF -S . -B out/Debug
+        if %errorlevel% neq 0 exit /b %errorlevel%
+
+        cmake --build out/Debug -j %NUMBER_OF_PROCESSORS%
+        if %errorlevel% neq 0 exit /b %errorlevel%
+
+        cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DANTLR_BUILD_CPP_TESTS=OFF -S . -B out/Release
+        if %errorlevel% neq 0 exit /b %errorlevel%
+
+        cmake --build out/Release -j %NUMBER_OF_PROCESSORS%
+        if %errorlevel% neq 0 exit /b %errorlevel%
+
+    - name: Build (non-Windows)
+      if: startswith(matrix.os, 'macos') || startswith(matrix.os, 'ubuntu')
+      run: |
+        if [ "${{matrix.compiler}}" == "clang" ]; then
+          export CC=clang
+          export CXX=clang++
+          echo 'CC=clang' >> $GITHUB_ENV
+          echo 'CXX=clang++' >> $GITHUB_ENV
+        else
+          export CC=gcc-9
+          export CXX=g++-9
+          echo 'CC=gcc-9' >> $GITHUB_ENV
+          echo 'CXX=g++-9' >> $GITHUB_ENV
+        fi
+
+        env
+        which cmake && cmake --version
+        which ninja && ninja --version
+        which $CC && $CC --version
+        which $CXX && $CXX --version
+
+        cd runtime/Cpp
+
+        cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DANTLR_BUILD_CPP_TESTS=OFF -S . -B out/Debug
+        cmake --build out/Debug --parallel
+
+        cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DANTLR_BUILD_CPP_TESTS=OFF -S . -B out/Release
+        cmake --build out/Release --parallel
+
+    - name: Prepare artifacts
+      if: always()
+      run: |
+        cd ${{ github.workspace }}/..
+        tar czfp antlr_${{ matrix.os }}_${{ matrix.compiler }}.tgz antlr4
```

Comment:
Is it possible to use more standard `zip` format instead of `tgz`?

Reply:
I am not sure if the default VM image for all platforms have zip installed or not. Will have to investigate.

BTW, _standard_ is very much platform dependent. On Windows, zip but on Posix it would be tar/tgz.